### PR TITLE
Change validation of put response hop limit to allow `1` as value to limit access to worker node's metadata endpoint

### PR DESCRIPTION
**Pull request by jakubbujny**
## what
* Change validation of `metadata_http_put_response_hop_limit` variable to allow to set `1` as value.

## why
* as standing in EKS best practises to limit access to worker node's metadata endpoint it's required to enable `metadata_http_tokens_required` and set `metadata_http_put_response_hop_limit` to 1 - see https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node

```
aws ec2 modify-instance-metadata-options --instance-id <value> --http-tokens required --http-put-response-hop-limit 1
```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -367,13 +367,12 @@ variable "metadata_http_put_response_hop_limit" {
   default     = 2
   description = <<-EOT
     The desired HTTP PUT response hop limit (between 1 and 64) for Instance Metadata Service requests.
-    The default is `2` to support containerized workloads.
```
Comment (Gowiem) [inline variables.tf:370]: Let's revert this line delete back unless you've got a reason for it. I see no reason to remove this additional bit of info.

**Review by Gowiem — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (jakubbujny) [reply variables.tf]: @Gowiem done

**Comment by christoffer-eide** [comment]
Any hope of getting this PR merged?

**Pull request closed without merging**